include/GL/glui.h

```diff
@@ -34,11 +34,23 @@
 #ifndef GLUI_GLUI_H
 #define GLUI_GLUI_H
 
-// Having stdlib here first fixes some 'exit() redefined' errors on MSVC.NET
-// that come from old GLUT headers.
-#include <cstdlib>
+//Note: GCC is not required for this feature. It's just an indicator. 
+#ifdef __GNUC__
+//GLUI_GLX uses glXGetCurrentDisplay to find GLUT's chosen X display.
+//It will #include GL/glx.h and X11/Xlib.h. This covers all desktops.
+#ifndef GLUI_GLX
+#define GLUI_GLX
+#endif
+#endif
+
+#if defined(GLUT_API_VERSION) //|| defined(?)
+
+  // This is for applications that do not want GLUT
+  // #included for them and it should be used if an
+  // alternative back-end is selected. 
+  typedef struct GLUquadric GLUquadric;
```

Comment:
Well the immediate change I did not because of abstraction but because `glui.h` was making things difficult for me, because I'm regularly switching between glut and freeglut right now, and GLUI is an optional (experimental) feature and is so not the mechanism by which GLUT is included in my code.

If building GLUI as a static library (which I recommend unless it's being packaged for distribution) it doesn't even need to know if it's using freeglut or GLUT. The goal anyway is not have `glui.h` contradict the application. If the application depends on freeglut's extensions then it can build GLUI to use them. If not GLUI should be GLUT agnostic. And I for one am not going to use GLUT long term, although I don't mind keeping it around as a back compatibility option.

It's very possible that GLUI will supersede GLUT. It could offer most of the same services but also things GLUT is unlikely ever to offer like Unicode input and output. If it does so then keeping the name "GLUI" attached to it could end up being a sly pun on "GLUT" of some kind.